packages/migrate/commonsChunkPlugin/commonsChunkPlugin.ts

```diff
@@ -0,0 +1,217 @@
+import {
+	addOrUpdateConfigObject,
+	createIdentifierOrLiteral,
+	createProperty,
+	findAndRemovePluginByName,
+	findPluginsByName,
+} from "@webpack-cli/utils/ast-utils";
+
+import { IJSCodeshift, INode } from "../types/NodePath";
+
+/**
+	*
+	* Transform for CommonsChunkPlugin. If found, removes the
+	* plugin and sets optimizations.namedModules to true
+	*
+	* @param {Object} j - jscodeshift top-level import
+	* @param {Node} ast - jscodeshift ast to transform
+	* @returns {Node} ast - jscodeshift ast
+	*/
+export default function(j: IJSCodeshift, ast: INode): INode {
+	const splitChunksProps: INode[] = [];
+	const cacheGroupsProps: INode[] = [];
+	const optimizationProps: object = {};
+
+	const commonCacheGroupsProps: INode[] = [
+		createProperty(j, "chunks", "initial"),
+		createProperty(j, "enforce", true),
+	];
+
+	// find old options
+	const CommonsChunkPlugin: INode = findPluginsByName(j, ast, [
+		"webpack.optimize.CommonsChunkPlugin",
+	]);
+
+	if (!CommonsChunkPlugin.size()) {
+		return ast;
+	}
+
+	// cache group options based on keys
+	let cacheGroup: object = {};
+	let cacheGroups: INode[] = [];
+	let isAsyncChunk: boolean = false;
+
+	// iterate each CommonsChunkPlugin instance
+	CommonsChunkPlugin.forEach(
+		(path: INode): void => {
+			const CCPProps: INode[] = path.value.arguments[0].properties;
+
+			// reset chunks from old props
+			cacheGroup = {};
+			cacheGroups = [];
+			isAsyncChunk = false;
+
+			let chunkKey: string;
+
+			// iterate CCP props and map SCP props
+			CCPProps.forEach(
+				(p: INode): void => {
+					const propKey: string = p.key.name;
+
+					switch (propKey) {
+						case "names":
+							p.value.elements.forEach((chunkName) => {
+								if (chunkName.value === "runtime") {
+									optimizationProps["runtimeChunk"] = createIdentifierOrLiteral( // tslint:disable-line
+										j,
+										true,
+									);
+								} else {
+									if (!Array.isArray(cacheGroup[chunkName.value])) {
+										cacheGroup[chunkName.value] = [];
+									}
+
+									if (chunkName.value === "vendor") {
+										cacheGroup[chunkName.value].push(
+											createProperty(j, "test", "/node_modules/"),
+										);
+									}
+								}
+							});
+							break;
+
+						case "name":
+							const nameKey = p.value.value;
+							if (nameKey === "runtime") {
+								optimizationProps["runtimeChunk"] = createIdentifierOrLiteral( // tslint:disable-line
+									j,
+									true,
+								);
+							} else {
+								chunkKey = nameKey;
+								if (!Array.isArray(cacheGroup[nameKey])) {
+									cacheGroup[nameKey] = [];
+								}
+								if (nameKey === "vendor") {
+									cacheGroup[nameKey].push(
+										createProperty(j, "test", "/node_modules/"),
+									);
+								}
+							}
+						 break;
+
+						case "async":
+							splitChunksProps.push(createProperty(j, "chunks", "async"));
+							isAsyncChunk = true;
+						 break;
+
```

Comment:
There also the `chunks` option.

Reply:
Can you please brief how that would work/map in SCP?